command/completion/completion.go

```diff
@@ -0,0 +1,107 @@
+package completion
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/flags"
+	"github.com/urfave/cli"
+	"go.step.sm/cli-utils/command"
+	"go.step.sm/cli-utils/errs"
+)
+
+func init() {
+	cmd := cli.Command{
+		Name:      "completion",
+		Usage:     "print the shell completion script",
+		UsageText: "**step completion** <shell>",
+		Description: `**step completion** command prints the shell completion script.
+
+## POSITIONAL ARGUMENTS
+
+<shell>
+: The shell program. Either bash or zsh.
+
+## EXAMPLES
+
+Add bash completion for the current user.
+'''
+$ step completion bash >> ~/.bash_completion
+'''
+`,
+		Flags: []cli.Flag{
+			flags.HiddenNoContext,
+		},
+		Action: Completion,
+		BashComplete: func(c *cli.Context) {
+			if c.NArg() > 0 {
+				return
+			}
+			fmt.Println("bash")
+			fmt.Println("zsh")
+		},
+	}
+
+	command.Register(cmd)
+}
+
+var bash = `# bash completion for step
```

Comment:
Can we read these values from the `./autocomplete/xxx` files? Otherwise we'll have to keep these updated in two places.

Reply:
There two options here:

1. Phase out the files in autocomplete and change all installation scripts to use the command instead.
2. Symlink the files inside the package and use something like embed to access them.

I'm leaning towards option 1. What do you think?

Reply:
I do like option 1 but I wonder if that will break other people's automation. I really don't know how people are installing `step` but if they expect those files to be present (as we do in our debian files and our Makefile) then they'll run into issues.

Reply:
We could keep them for a while and transition everything towards the command, and at some point later remove them. The completion scripts are very unlikely to change, so we wouldn't worry about them diverging.

Reply:
I like that idea. 

The places that I think we need to update:
* .goreleaser.yml
* debian/step-cli.bash-completion
* make/bundle.sh

Reply:
We can add a README in the autocomplete dir that says those files are deprecated and the source of truth in the future will be `step completion <shell>`.

Reply:
Sounds good. Should I do them in this PR?

Reply:
Please, if you wouldn't mind. I know they're not "directly related" to your initial issue.

Reply:
It's ready now. @dopey